src/System.Private.CoreLib/src/System/Runtime/CompilerServices/RuntimeHelpers.cs

```diff
@@ -197,6 +197,26 @@ public static bool IsReferenceOrContainsReferences<T>()
             throw new InvalidOperationException();
         }
 
+        /// <summary>
+        /// GetArrayRange helper method for the compiler to slice an array using a range.
+        /// </summary>
+        public static T[] GetArrayRange<T>(T[] array, Range range)
```

Comment:
@terrajobst could you please comment on this one too? I just picked the name from your spec and I agree with @agocke the name GetArrayRange doesn't look right. should we call this one Slice too?

Reply:
My care level is low. I think the name was originally proposed by the compiler team. Since we don't expect users to call it, almost any name works for. `GetSubArray()` works for me :-)